File: java/src/org/iota/wasplib/contracts/HelloWorld.java
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package org.iota.wasplib.contracts;

import org.iota.wasplib.client.context.ScCallContext;
import org.iota.wasplib.client.exports.ScExports;

public class HelloWorld {
	//export on_load
	public static void onLoad() {
		ScExports exports = new ScExports();
		exports.AddCall("hello_world", HelloWorld::helloWorld);
	}

	public static void helloWorld(ScCallContext sc) {
		sc.Log("Hello, world!");
	}
}

File: rust/contracts/helloworld/test/helloworld/wasm/helloworld.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

// +build wasm

package main

import (
	"github.com/iotaledger/wasplib/rust/contracts/helloworld/test/helloworld"
	"github.com/iotaledger/wasplib/client/wasm"
)

func main() {
}

//export on_load
func helloworldOnLoad() {
	wasmclient.ConnectWasmHost()
	helloworld.OnLoad()
}
